barretenberg/cpp/src/barretenberg/vm/avm/trace/trace.cpp

```diff
@@ -3771,9 +4097,9 @@ ReturnDataError AvmTraceBuilder::op_revert(uint8_t indirect, uint32_t ret_offset
     pc = is_top_level ? UINT32_MAX : current_ext_call_ctx.last_pc;
     auto return_data = is_top_level ? returndata : current_ext_call_ctx.nested_returndata;
 
-    if (is_ok(error)) {
-        error = AvmError::REVERT_OPCODE;
-    }
+    // if (is_ok(error)) {
+    //     error = AvmError::REVERT_OPCODE;
+    // }
 
```

Comment:
Treating errors as exceptional halts that consume all gas. Explicit reverts do not. I could alternatively this error type in execution.cpp in a special way.